extends Area3D

@export var page_index: int = 3  # NumÃ©ro de la page (0 Ã  8)
@onready var audio_player: AudioStreamPlayer3D = $AudioStreamPlayer3D

func _ready():
	connect("body_entered", Callable(self, "_on_body_entered"))

func _on_body_entered(body):
	if not body.is_in_group("player"):
		return

	if page_index in Global.pages_trouvees:
		return

	Global.pages_trouvees.append(page_index)
	print("ğŸ“˜ Page trouvÃ©e :", page_index, "| Total :", Global.pages_trouvees.size(), "/ 9")

	play_random_sound()

	# Supprime la page aprÃ¨s le son
	await get_tree().create_timer(1).timeout
	queue_free()

func play_random_sound():
	var random_id = randi_range(1, 4)
	var sound_path = "res://assets/audio/doublage/TrouvÃ©!" + ("" if random_id == 1 else str(random_id)) + ".wav"

	var sound = load(sound_path)
	if sound:
		audio_player.stream = sound
		audio_player.play()
	else:
		print("âš ï¸ Fichier introuvable :", sound_path)
